docs/advanced-features/rust-compiler.md

```diff
@@ -0,0 +1,139 @@
+---
+description: Learn about the Rust Compiler, built with SWC, which transforms and minifies your Next.js application.
+---
+
+# Rust Compiler
+
+<details open>
+  <summary><b>Version History</b></summary>
+
+| Version   | Changes                                                      |
+| --------- | ------------------------------------------------------------ |
+| `v12.0.0` | Rust Compiler [introduced](https://nextjs.org/blog/next-12). |
+
+</details>
+
+The Rust Compiler for Next.js, built with [SWC](http://swc.rs/), allows Next.js to transform and minify your code for production. This replaces Babel for individual files and Terser for minifying output bundles.
```

Comment:
```suggestion
The Rust Compiler for Next.js, built with [SWC](http://swc.rs/), allows Next.js to transform and minify your JavaScript and Typescript code for production. This replaces Babel for individual files and Terser for minifying output bundles.
```

This might make it more clear you're not compiling rust code

Reply:
Agreed!